libraries/omnibus_build.rb

```diff
@@ -16,163 +16,125 @@
-      ].join(' ')
+    #######################################
+    # helpers for default property settings
+    #######################################
+    def default_install_dir
+      Chef::Platform.windows? ? ::File.join(ENV['SYSTEMDRIVE'], project_name) : "/opt/#{project_name}"
     end
 
-    def prepare_build_enviornment
-      # Optionally wipe all caches (including the git cache)
-      if new_resource.expire_cache
-        cache = Resource::Directory.new(new_resource.base_dir, run_context)
-        cache.recursive(true)
-        cache.run_action(:delete)
-      end
-
-      # Clean up various directories from the last build
-      %W(
-        #{new_resource.base_dir}/build/#{new_resource.project_name}/*.manifest
-        #{new_resource.base_dir}/pkg
-        #{new_resource.project_dir}/pkg
-        #{new_resource.install_dir}
-      ).each do |directory|
-        d = Resource::Directory.new(directory, run_context)
-        d.recursive(true)
-        d.run_action(:delete)
-      end
-
-      # Create required build directories with the correct ownership
-      %W(
-        #{new_resource.base_dir}
-        #{new_resource.install_dir}
-      ).each do |directory|
-        d = Resource::Directory.new(directory, run_context)
-        d.owner(new_resource.build_user)
-        d.run_action(:create)
-      end
+    def default_base_dir
+      Chef::Platform.windows? ? ::File.join(ENV['SYSTEMDRIVE'], 'omnibus-ruby') : '/var/cache/omnibus'
     end
 
-    def execute_with_omnibus_toolchain(command)
-      execute = Resource::Execute.new("#{new_resource.project_name}: #{command}", run_context)
-      execute.command(
-        <<-CODE.gsub(/^ {10}/, '')
-          . #{::File.join(build_user_home, 'load-omnibus-toolchain.sh')}
-          #{command}
-        CODE
-      )
-      execute.cwd(new_resource.project_dir)
-      execute.environment(environment)
-      execute.user(new_resource.build_user)
-      execute.run_action(:run)
+    def default_config_file
+      ::File.join(project_dir, 'omnibus.rb')
     end
 
-    def environment
-      environment = new_resource.environment || {}
-      # Ensure we inheriet the calling procceses $PATH
-      environment['PATH'] = ENV['PATH']
-      # We need to all underlying build process respect the build user
-      # as specified by the `build_user` attribute.
-      environment['USER']     = new_resource.build_user
-      environment['USERNAME'] = new_resource.build_user
-      environment['LOGNAME']  = new_resource.build_user
-      # Ensure we don't inherit the $TMPDIR of the calling process. $TMPDIR
-      # is set per user so we can hit permission issues when we execute
-      # the build as a different user.
-      environment['TMPDIR'] = nil if mac_os_x?
-      environment
+    def default_environment
+      {
+        'PATH' => ENV['PATH'],
+        'USER' => build_user,
+        'USERNAME' => build_user,
+        'LOGNAME' => build_user,
+        'TMPDIR' => nil
+      }
```

Comment:
The previous `environment` method was not just default values. The `PATH`, `USER`, `USERNAME`, `LOGNAME` and `TMPDIR` values are required for every forked process or builds will fail on many older Unixes _and_ Mac OS X. Please restore the previous logic including comments which explain the purpose of each env var we set.